edb/ir/typeutils.py

```diff
@@ -196,6 +196,28 @@ def is_persistent_tuple(typeref: irast.TypeRef) -> bool:
         return False
 
 
+def get_custom_serialization(
+    typeref: irast.TypeRef,
+) -> Optional[str]:
+    # FIXME: instead of hardcode we need to extract this from the
+    # schema/extension
+    if typeref.real_base_type.id in {
+        s_obj.get_known_type_id('ext::postgis::box2d'),
+        s_obj.get_known_type_id('ext::postgis::box3d'),
+    }:
+        return 'geometry'
+
+    return None
```

Comment:
We should add a `custom_serialization` field to `ScalarType`, I guess.

(We should skip using known ids if we are temporarily hacking around it. Can we use `name_hint`?)